ui/src/reducers/roles.js

```diff
@@ -23,5 +24,7 @@ export default createReducer({
 
   [updateRole.COMPLETE]: (state, { id, data }) => objectLoadComplete(state, id, data),
 
+  [fetchMetadata.COMPLETE]: (state, { metadata }) => {  if (!metadata?.token) { return state } else { return handleLogin(state, metadata.token);}},
```

Comment:
Is this duplicative with `session.js`?

Reply:
yes, this is duplicate because `handleLogin` is duplicative.

Reply:
deduplicated